### PR TITLE
Feature/test coverage fix

### PR DESCRIPTION
**Pull request by iRhysBradbury**

### CODE REVIEW

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package uk.gov.ons.addressIndex.server.modules
```
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: Those things are already tested, is there a need for additional ones?

Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: These tests are unit tests for the QueryDefinition objects.
The other tests are for the response from ES.

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala [hunk 2]
```diff
@@ -111,6 +118,12 @@ class AddressControllerSpec  extends PlaySpec with Results with AddressIndexCann
       Future.successful(HybridAddresses(Seq.empty, 1.0f, 0))
 
     override def client(): ElasticClient = ElasticClient.local(Settings.builder().build())
+
+    override def logger: Logger = ???
```
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: I think adding additional public methods to the interface can be avoided.
The user of the interface don't need to know that it has logger, neither does he need to know that there are query generation methods as the user of the interface should never use them.

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala [hunk 3]
```diff
@@ -16,27 +16,43 @@ import scala.concurrent.{ExecutionContext, Future}
...
+    * @param info
+    * @param t
+    * @param executable
+    * @tparam T
+    * @tparam R
+    * @tparam Q
+    * @return
     */
-  def queryAddresses(start: Int, limit: Int, tokens: Seq[CrfTokenResult]): Future[HybridAddresses]
+  def logExecute[T, R, Q](info: String)(t: T)(implicit executable: Executable[T, R, Q]): Future[Q] = {
```
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: I saw this method in your PR, while it's neat, after a careful consideration, I would suggest to go without it as the code is more explicit (casting ES to DTO) this way.

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: I'm not sure what you're asking; could you rephrase?

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: This method doesn't cast from ES to DTO.

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: Those methods shouldn't be public as the user of the interface shouldn't use them.

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala (hunk 1 above)
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: Good spot, the more tests the merrier?

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: Yes, it also logs the query, but this functionality is already implemented directly in the method

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: `logger` is used in `logExecute`
@AlWestVT mentioned he'd like to see more logs

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala (hunk 1 above)
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: The one linked is complete (all tokens are provided), if you want to provide one where not all tokens are provided (I'd say that we can avoid doing it as query modification already requires us to modify corresponding 1 test), you may add them near the linked test in the same manner (the expected and the results are Json object instead of Strings)

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: We already log queries
https://github.com/ONSdigital/address-index-api/blob/develop/server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala#L80
(and the same for query by uprn)

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala (hunk 1 above)
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: I'm not sure what you're asking here - would you like me to remove these tests?

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (iRhysBradbury) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: I'm not sure what you're asking here - would you like me to remove these logs?

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: I'm not sure what you're asking here - would you like me to remove these logs?

File: server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala (hunk 1 above)
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/modules/ElasticsearchRepositorySpec2.scala]: I'm saying that I already implemented those tests and there is no need in additional ones in my opinion.

File: server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala (hunk 2 above)
Comment (Mironor) [inline server/test/uk/gov/ons/addressIndex/server/controllers/AddressControllerSpec.scala]: I'm saying that the functionality is already implemented and there is no need in refactoring it now in my opinion.

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: I'm saying that the functionality is already implemented and there is no need in refactoring it now in my opinion.
